File: sdks/python/sealstack/namespaces/connectors.py
"""Connectors namespace (read-only)."""

from typing import Any

from .._http import HttpClient


class ConnectorsNamespace:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def list(self) -> list[Any]:
        out = await self._http.request("GET", "/v1/connectors")
        return out["connectors"]

File: sdks/typescript/src/namespaces/connectors.ts
import type { HttpClient } from "../http.js";

export class ConnectorsNamespace {
  constructor(private http: HttpClient) {}

  async list(): Promise<unknown[]> {
    const data = await this.http.request<{ connectors: unknown[] }>({
      method: "GET", path: "/v1/connectors",
    });
    return data.connectors;
  }
}
